src/nodes/lighting/ShadowNode.js

```diff
@@ -662,12 +662,21 @@ class ShadowNode extends ShadowBaseNode {
 		const currentRenderObjectFunction = renderer.getRenderObjectFunction();
 		const currentMRT = renderer.getMRT();
 
+		const useVelocity = currentMRT ? currentMRT.has( 'velocity' ) : false;
+
 		renderer.setMRT( null );
 
 		renderer.setRenderObjectFunction( ( object, scene, _camera, geometry, material, group, ...params ) => {
 
 			if ( object.castShadow === true || ( object.receiveShadow && shadowType === VSMShadowMap ) ) {
 
+				if ( useVelocity ) {
+
+					object.userData = object.userData || {};
+					object.userData.useVelocity = true;
```

Comment:
May be better to do a local `WeakMap` for this?

Reply:
Oh, I can see that it's being accessed from another class...
Maybe better to call it `_useVelocity` then?